subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -368,65 +364,6 @@ public void execute(final Test test) {
                 configureTestDefaults(test, project, convention);
             }
         });
-        project.getGradle().getTaskGraph().whenReady(new Action<TaskExecutionGraph>() {
```

Comment:
what was missing from the lazy task API that meant we needed to restructure all of this?

Reply:
This was trying to run a bit of configuration after all other configuration by waiting until the execution graph had been decided.  This is the only case of this I've seen. We were reconfiguring some properties based on system properties right before execution.  This was safe for what we were doing, but it wouldn't be in general.

It looks like this now works the way it was with a `configureEach` added in the `whenReady` callback.  I suspect it might be due to Gary's refactorings. Previously, this was causing all `Test` tasks to be configured.

I think using command-line options is the better solution for this particular feature (`--debug-jvm` vs -Dtest.debug`) and does the right thing out of the box (command-line options are applied after everything else).